guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-block.md

```diff
@@ -0,0 +1,269 @@
+{% endraw %}
+{% endcode %}
+
+And let's add a bit of CSS code (not neccessary)
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/preview/sw-cms-preview-button.scss" %}
+```css
+.ninja-button-flex-center {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 1rem;
+    .sw-cms-preview-ninja-button {
+        background-color: #4492ed;
+        border: none;
+        color: white;
+        padding: 16px 60px;
+        text-align: center;
+        text-decoration: none;
+        display: inline-block;
+        font-size: 16px;
+    }
+
+}
+```
+{% endcode %}
+
+### Build administration
+
+You need to build your administration before you can see anything in the shopware backend. Just run the one of the following commands:
+
+{% tabs %}
+{% tab title="Development template" %}
+```bash
+./psh.phar administration:build
+```
+{% endtab %}
+
+{% tab title="Production template" %}
+```bash
+./bin/build-administration.sh
+```
+{% endtab %}
+{% endtabs %}
+
+Note: Your plugin has to be installed and activated. If you still can't see the block make sure to clear your cache as well.
+
+## What about the slots?
+
+If you did everything correct so far, you should see the CMS-Block in your "Shopping Experiences" module in the category text. You also should be able
+to drag and drop the block into your CMS-Page but you will not see a button there. Instead you will see the exact same "Lorem Ipsum" text like you do
+when you drag and drop a normal text block into your page. Why?
+
+Because of the slot we defined. In our `<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/index.js`. We gave the slot
+the element "text". You can throw in every other element here as well. Since we haven't created a element for the button it's one of our To-Do's now.
+
+We also need a config and we need to react on the settings which are made in the administration to display everything correct in the frontend. First of all
+we need to create a simple frontend view first. So let's start with that!
+
+## Frontend implemantation
+
+Creating a frontend view is very simple as you should know from the [Add CMS block](../add-cms-block.md) guide. We only need to create a twig file in the 
+correct directory with the same name. It should look like this:
+
+
+{% code title="<plugin root>/src/Resources/views/storefront/block/cms-block-cms-button.html.twig" %}
+{% raw %}
+```text
+{% block block_cms_button_example %}
+    <div class="cms-element ninja-cms-button-flex ninja-cms-button-flex-center">
+		{% block element_text_button %}
+			<a href="#">
+				<button class="ninja-cms-btn">
+                    Button
+                </button>
+			</a>
+		{% endblock %}
+    </div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+Mayb you remember the storefront implementation from the [Add CMS block](../add-cms-block.md) guide. In that guide we were just extending from
```

Comment:
"Mayb*e* you remember" sounds colloquial

Maybe "As previously mentioned in" is better?

Reply:
"As previously mentioned in" is better I guess. First time writing a guide.. Let me change that!

Reply:
Nice that you started here :) I think here you will get valuable feedback. Not just from a random non-shopware-org-member like me :D